runtime/etc/Makefile.mli-shared

```diff
@@ -67,16 +62,26 @@ $(TMPBINNAME): $(CHPL_CL_OBJS) FORCE
 #
 # Compile the client bundle. Note the use of $(SHARED_LIB_CFLAGS) here.
 #
+ifdef CHPL_ZMQ_HOME
```

Comment:
If we can eliminate the duplication in the else branch that's cool, but if not, I'm not hung up about it. Makefiles are ugly to begin with. We can rest easy knowing that we have the best formatted Makefiles in town.

Reply:
No dice on the inner `ifdef/else` :(

Reply:
For all of the repeated patterns below, I'd strongly suggest taking the following approach:

Near the top of your Makefile, define variables conditionally, e.g.:

```make
ifdef CHPL_ZMQ_HOME
CHPL_MLI_INCS = -I$(CHPL_ZMQ_LOC)/include
...
endif
```

and then just have a single rule for the commands themselves that use these variables (which will default to nothing if they were not defined):

```make
	$(CC) $(CHPL_MAKE_BASE_CFLAGS) $(GEN_CFLAGS) $(COMP_GEN_CFLAGS) \
		$(SHARED_LIB_CFLAGS) \
		-c -o $(TMPBINNAME).o \
		$(CHPLSRC) \
		$(COMP_GEN_MLI_EXTRA_INCLUDES) \
		$(CHPL_RT_INC_DIR) \
		$(CHPL_MLI_INCS)
```

This is consistent with what we do throughout our Makefiles (rely on variables not being set becoming empty strings) and eliminates repetition that's going to lead someone to inevitably change one branch but not the other.

Alternatively and equivalently beneficially, the conditional could just modify an existing variable rather than defining a new one, e.g.:

```make
ifdef CHPL_ZMQ_HOME
CHPL_RT_INC_DIR += -I$(CHPL_ZMQ_LOC)/include
...
endif
```

Reply:
Oh, fair point!

Reply:
I just opened #13122 to do this.  At least one of the new Makefile variable names is probably horrible, but I'm a little braindead right now so I'm going to step away from it for a sec